src/components/EntityInfo.vue

```diff
@@ -316,10 +301,7 @@ export default class EntityInfo extends Mixins(CommonMixin, DateMixin, EnumMixin
 
   /** Redirects the user to the Auth UI to update their business profile. */
   private editBusinessProfile (): void {
-    // safety checks
-    if (!this.hasBlocker && !this.isHistorical) {
```

Comment:
No need for safety checks: it's now clear in the the template that we can't get here unless allowed.